osu.Game.Rulesets.Tau/Objects/Drawables/DrawableSlider.cs

```diff
@@ -141,13 +147,20 @@ protected override void ClearNestedHitObjects()
         private float convertNoteSizeToSliderSize(float beatSize)
             => Interpolation.ValueAt(beatSize, 2f, 7f, 10f, 25f);
 
+        private Color4 convertHighlightSettingToColor(bool highlightHardBeats)
+            => highlightHardBeats && HitObject.IsHard ? Color4.Orange : Color4.White;
+
         [BackgroundDependencyLoader]
-        private void load(IRenderer renderer, GameHost host)
+        private void load(IRenderer renderer, GameHost host, TauRulesetConfigManager config)
         {
+            config.BindWith(TauRulesetSettings.HighlightHardBeats, HighlightHardBeats);
+
             NoteSize.BindValueChanged(value => path.PathRadius = convertNoteSizeToSliderSize(value.NewValue), true);
+            // HighlightHardBeats.BindValueChanged(value => updatePathTexture(value.NewValue, renderer));
 
             host.DrawThread.Scheduler.AddDelayed(() => drawCache.Invalidate(), 0, true);
-            path.Texture = properties.SliderTexture ??= generateSmoothPathTexture(renderer, path.PathRadius, _ => Color4.White);
+            pathTexture = generateSmoothPathTexture(renderer, path.PathRadius, _ => Color4.White);
+            pathTextureHighlighted = generateSmoothPathTexture(renderer, path.PathRadius, _ => Color4.Orange);
```

Comment:
It seems odd to always generate both textures at once, when the most people aren't toggling the setting during gameplay.

Would it be feasible to only generate the texture only when it is needed? Something along the lines of the following, but cleaner ofc.
```cs
HighlightHardBeats.BindValueChanged(value => path.Texture = ((value.NewValue && HitObject.IsHard) ? 
    pathTextureHighlighted ??= generateSmoothPathTexture(renderer, path.PathRadius, _ => Color4.Orange) : 
    pathTexture ??= generateSmoothPathTexture(renderer, path.PathRadius, _ => Color4.White);, true));
```